src/Compilers/CSharp/Test/Emit/Emit/CompilationEmitTests.cs

```diff
@@ -935,12 +1203,30 @@ I1<I2> M(A a)
         }
 
         [Fact]
+        public void RefAssemblyClient_EmitNestedTypes()
+        {
+            VerifyRefAssemblyClient(@"
+public class A
+{
+    public class Nested { }
+}
+",
+@"class C
+{
+    void M(A.Nested a) { }
+}",
+comp => comp.VerifyDiagnostics());
+        }
+
+        [Fact]
         public void RefAssemblyClient_StructWithPrivateGenericField()
```

Comment:
Do those tests test a type that is private to the reference assembly? For example:
```
struct S
{
    private class C { }
    private C F;
}
```